### PR TITLE
6.0 update models

### PR DESCRIPTION
**Pull request by kimc0de**

### CODE REVIEW

File: controllers/candidatesController.js [hunk 1]
```diff
@@ -1,93 +1,86 @@
...
+    //CREATE
+    create: (req, res, next) => {
+		let candidateParams = {
+            preferred_position: req.body.preferred_position,
+            soft_skills: req.body.soft_skills,
+            other_aspects: req.body.other_aspects,
+            work_culture_preferences: req.body.work_culture_preferences,
+        }
+		Candidate.create(candidateParams)
+			.then(candidate => {
```
Comment (joshuameiser) [inline controllers/candidatesController.js:37]: Why did you delete the flash message?

File: controllers/candidatesController.js [hunk 2]
```diff
@@ -1,93 +1,86 @@
...
-      })
-      .catch(error => {
-        console.log(`Error saving candidate profile: ${error.message}`);
-        res.locals.redirect = "/candidates/new";
-        req.flash(
-          "error",
-          `Failed to create user account because: ${error.message}.`
-        );
-        next();
+        console.log('a candidate saved into both MongoDB and Elasticsearch')
```
Comment (joshuameiser) [inline controllers/candidatesController.js:40]: Once again, why did you delete the flash message & in this case change the structure?
Also, why was the redirect deleted?

File: controllers/candidatesController.js [hunk 3]
```diff
@@ -98,29 +91,33 @@ module.exports = {
...
-      .then(candidate => {
-        res.locals.redirect = `/user/${req.app.locals.user._id}`;
-        res.locals.candidate = candidate;
-        next();
+    //has to be findOneAndUpdate bc it works with mongoosastic
+    Candidate.findOneAndUpdate({_id: candidateId},
+      {$set: candidateParams},
+      {new: true})
+    .then(candidate => {
+      res.locals.redirect = `/candidates/${candidateId}`;
```
Comment (joshuameiser) [inline controllers/candidatesController.js:99]: We changed the redirect for a reason. That shouldn't be changed

File: models/candidate.js [hunk 4]
```diff
@@ -7,21 +7,33 @@ const mongoose = require("mongoose"),
...
+        soft_skills: [{
+          type : String,
+          required : true
+        }],
         other_aspects: [String],
         work_culture_preferences: [{
-            type: String,
-            required: true
+          name: String,
+          importance: {
```
Comment (joshuameiser) [inline models/candidate.js:31]: Did we really want to have an importance for work_culture_preferences? I thought we agreed on only having importance for hard skills
Might be mistaken here though

File: models/job_offer.js [hunk 5]
```diff
@@ -18,8 +18,24 @@ jobSchema = mongoose.Schema({
...
+    }],
+    hard_skills: [{
+      name: String,
+      importance: {
+        type: Number,
+        min : 1,
+        max : 4,
+        required : true
+      }
+    }],
```
Comment (joshuameiser) [inline models/job_offer.js:38]: Shouldn't there also be something similar to work_culture_preferences in job_offer? I thought we wanted both to have the same fields more or less

**Review by joshuameiser — CHANGES_REQUESTED** [review]
I don't see why the controller has been changed. We implemented flash messages for a reason. Also, there's one redirect that has been changed back - though we needed the one that was deleted

File: controllers/candidatesController.js (hunk 3 above)
Comment (kimc0de) [reply controllers/candidatesController.js:99]: hey.. sorry about the controller.. We actually didnt change anything there but might because when we do the pull rebase, there was some mistake with merging code.. I will fix that. Thanks a lot for your review.

File: models/candidate.js (hunk 4 above)
Comment (kimc0de) [reply models/candidate.js:31]: We did want the importance of work culture in candidate.js but not in job_offer bc the work culture is what the recruiter provides
